### ISSUE TITLE
"guest user can set displaynames" is flaky

### ISSUE THREAD

**Issue by richvdh** [matrix-org/sytest#819]
eg https://buildkite.com/matrix-dot-org/synapse/builds/7348#f0749e42-bda8-48f9-95fb-3f862aa7fca4

```
# Started: 2020-02-26 09:55:08.990
# Ended: 2020-02-26 09:55:10.021
# Got 7, expected "creeper" for Room displayname at tests/30rooms/13guestaccess.pl line 190.
# 0.167456: Registered new user @anon-20200226_095239-167:localhost:8831
# 0.919001: User @7:localhost:8831 joined room
# { room_id => "!QgjZoiPzZxNvhctfuI:localhost:8831" }
```

**Comment by clokep** [comment]
I saw a couple more instances of this, both on worker runs.

The assertion that fails is:

https://github.com/matrix-org/sytest/blob/128d7d31900abc02767c7572ed19bb88c37590a7/tests/30rooms/13guestaccess.pl#L190

**Comment by clokep** [comment]
Been able to reproduce this locally (although it seems to happen only 1 out of every 4 times or so):

```
docker run --rm -it -e POSTGRES=true -e WORKERS=true -v /Users/clokep/matrix/synapse\:/src:ro -v /Users/clokep/matrix/sytest/logs\:/logs -v /Users/clokep/matrix/sytest\:/sytest:ro matrixdotorg/sytest-synapse:py35 tests/30rooms/13guestaccess.pl
```

**Comment by clokep** [comment]
I attempted to fix this (see 5b1960d727bda09ef980893a70baf2927b48c003), but that seems to have caused a bunch of other tests to have gone flaky.

**Comment by clokep** [comment]
I think the failed tests on ^ were the other failures recently fixed. I rebased and that passed without issue.
